app/src/main/java/org/schabi/newpipe/error/ReCaptchaActivity.java

```diff
@@ -216,17 +216,21 @@ private void addYoutubeCookies(@NonNull final String cookies) {
         if (cookies.contains("s_gl=") || cookies.contains("goojf=")
                 || cookies.contains("VISITOR_INFO1_LIVE=")
                 || cookies.contains("GOOGLE_ABUSE_EXEMPTION=")) {
-            // youtube seems to also need the other cookies:
-            addCookie(cookies);
+            // Remove CONSENT cookie; otherwise it will be set twice
+            // YouTube seems to need the other cookies
+            addCookie(cookies
+                    .replaceAll("CONSENT=.*?;\\s", "") // cookie in the middle
+                    .replaceAll(";\\sCONSENT=.*", "")); // cookie at the end
```

Comment:
Is it possible to unit test this? From the looks of it one could test `handleCookies`, if that was extracted out of the activity. And through that test the behaviour that was changed here

Reply:
unfortunately this doesn't work for me